### PR TITLE
Allow same param to be used for multiple properties

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/main.bicep [hunk 1]
```diff
@@ -0,0 +1,21 @@
+targetScope = 'subscription'
+
+@description('')
+param overrideLocation string
+
+
+resource resourceGroup_I6QNkoPsb 'Microsoft.Resources/resourceGroups@2023-07-01' = {
+  name: 'rg-TEST'
+  location: 'westus'
```
Comment (JoshLove-msft) [inline sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/main.bicep:9]: hmm this should be using overrideLocation

Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/main.bicep:9]: no.. this is for the RG

File: sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/resources/rg_TEST_module/rg_TEST_module.bicep [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@description('')
+param overrideLocation string
+
+
+resource storageAccount_melvnlpF2 'Microsoft.Storage/storageAccounts@2022-09-01' = {
+  name: toLower(take(concat('photoAcct', uniqueString(resourceGroup().id)), 24))
+  location: 'westus'
```
Comment (JoshLove-msft) [inline sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/resources/rg_TEST_module/rg_TEST_module.bicep:7]: This should be using the param

Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/tests/Infrastructure/CanAssignParameterToMultipleResources/resources/rg_TEST_module/rg_TEST_module.bicep:7]: oh nevermind..

**Review by m-nash — APPROVED** [review]

**Pull request merged**